R/class-workbook.R

```diff
@@ -289,6 +290,29 @@ wbWorkbook <- R6::R6Class(
       self
     },
 
+    #' @description validate sheet
+    #' @param sheet A character sheet name or integer location
+    #' @returns The integer position of the sheet
+    validate_sheet = function(sheet) {
+
+      # workbook has no sheets
+      if (is.null(self$sheet_names)) {
```

Comment:
maybe `length()`? guess if we remove the only sheet, we end up with `character()`